server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -2144,6 +2152,38 @@ private boolean checkForNonStoppedVmInNetwork(long networkId) {
         return vms.isEmpty();
     }
 
+    private void replugNicsForUpdatedNetwork(NetworkVO network) throws ResourceUnavailableException, InsufficientCapacityException {
+        List<NicVO> nics = _nicDao.listByNetworkId(network.getId());
+        Network updatedNetwork = getNetwork(network.getId());
+        for (NicVO nic : nics) {
+            long vmId = nic.getInstanceId();
+            VMInstanceVO vm = _vmDao.findById(vmId);
+            if (vm == null) {
+                s_logger.error("VM for nic " + nic.getId() + " not found with Vm Id:" + vmId);
+                continue;
+            }
+            if (!Hypervisor.HypervisorType.VMware.equals(vm.getHypervisorType())) {
+                s_logger.debug("VM is not on VMware");
```

Comment:
Maybe add more information to the following log lines? Something like: `Cannot replug nic <ID> as: VM ...`